pkg/controllers/pushsecret/pushsecret_controller.go

```diff
@@ -465,3 +472,40 @@ func statusRef(ref v1beta1.PushSecretData) string {
 	}
 	return ref.GetRemoteKey()
 }
+
+// removeUnmanagedStores iterates over all SecretStore references and evaluates the controlerClass property.
+// Warns for all unmanaged stores.
+// Returns a map containing only managed stores.
+func removeUnmanagedStores(ctx context.Context, namespace string, r *Reconciler, ss map[esapi.PushSecretStoreRef]v1beta1.GenericStore, ps esapi.PushSecret) (map[esapi.PushSecretStoreRef]v1beta1.GenericStore, error) {
+	for ref := range ss {
+		var store v1beta1.GenericStore
+		switch ref.Kind {
+		case v1beta1.SecretStoreKind:
+			store = &v1beta1.SecretStore{}
+		case v1beta1.ClusterSecretStoreKind:
+			store = &v1beta1.ClusterSecretStore{}
+			namespace = ""
+		}
+		err := r.Client.Get(ctx, types.NamespacedName{
+			Name:      ref.Name,
+			Namespace: namespace,
+		}, store)
+
+		if err != nil {
+			return ss, err
+		}
+
+		class := store.GetSpec().Controller
+		if class != "" && class != r.ControllerClass {
+			warnMsg := fmt.Sprintf("Controller for the store %q does not match the reconciler's controller %q, so the store %q is not managed by the controller %q", class, r.ControllerClass, ref.Name, r.ControllerClass)
+			r.recorder.Event(&ps, v1.EventTypeWarning, esapi.ReasonErrored, warnMsg)
+			delete(ss, ref)
+		}
+	}
+
+	if len(ss) == 0 {
```

Comment:
Isn't it okay if there are no unmanaged stores?

Reply:
Yes, I think it should be fine to skip if there are no unmanaged stores, but this is different to what @gusfcarvalho originally asked for. Should I change the code to skip then? I'm not sure what the test would look like though.

Reply:
Err maybe just rename the variable? That's what confused me. Also yeah skip then the rest of the logic works better. You're right.

Reply:
https://github.com/external-secrets/external-secrets/pull/3123/files#diff-0761d75b52f80b7ba910fa84aaa8eff22de970bfac4cbad894a4e966b0c6dc8dR1251

Pushed my code with the skip logic, but not sure how to test it - could you give some guidance please. I imagine the controller in the test here sees that it has a PushSecret with an unmanaged store, but I'm not sure if it would try and update the status of that store.

In our real use case, we have a `devops` controller and a `default` controller and one manages the PushSecret, so the PushSecret status would be synced successfully. However, I'm not sure what to do in the tests here, as there is only one controller.

Reply:
You could test it by checking that `PushSecretToProviders` was not called.

Reply:
You can check what we mock in that flow. Like `Get` or `SecretExists` or `PushSecret`.

Reply:
For now I've just tested that conditions is empty as that seems like the simplest - the controller shouldn't touch the PS and so, it has no conditions. Is that ok?

Reply:
Yep, works perfectly. Thank you 🙇

Reply:
Awesome, done. I think PR is ready for review again

Reply:
Sorry, git blunder again! I think PR is ready for final review, unless I've missed something (I have found the GH workflow quite confusing to work with...)